scene/gui/text_edit.cpp

```diff
@@ -6157,16 +6177,64 @@ void TextEdit::set_code_hint(const String &p_hint) {
 	update();
 }
 
-void TextEdit::code_complete(const Vector<String> &p_strings, bool p_forced) {
+void TextEdit::code_complete(const List<ScriptCodeCompletionOption> &p_strings, bool p_forced) {
 
-	completion_strings = p_strings;
+	completion_sources = p_strings;
 	completion_active = true;
 	completion_forced = p_forced;
-	completion_current = "";
+	completion_current = ScriptCodeCompletionOption();
 	completion_index = 0;
 	_update_completion_candidates();
 }
 
+Ref<Texture> TextEdit::get_completion_icon(const ScriptCodeCompletionOption &p_option) {
+	Ref<Texture> tex = NULL;
+#ifdef TOOLS_ENABLED
```

Comment:
I'm not sure that adding references to editor icons in TextEdit is the right approach, see #29730.
Could this be done via CodeTextEditor instead? (cc @Paulb23)

Reply:
Completion options are drawn by `TextEdit` but the `CodeTextEditor` doesn't inherited from `TextEdit` so I have to put it in `TextEdit`.